Angular/src/app/auth-guard/auth-guard.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@angular/core";
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from "@angular/router";
+import { CookieService } from 'ngx-cookie-service';
+import privateKey from '../../../../privateKey.js'
+import { decode, verify } from 'jsonwebtoken'
+
+
+@Injectable()
+export class AuthGuardComponent implements CanActivate {
+
+    private token: Object
+
+    constructor(private router: Router, private cookieService: CookieService){}
+    
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
+      var token = this.cookieService.get('token')
+      if(token != undefined && token != null && token != ''){
```

Comment:
Do you need to check if `token` is undefined or null here?
I think `this.cookieService.get` will always return a cookie (string) or an empty string, so you may only need `if (token) {` here (https://developer.mozilla.org/en-US/docs/Glossary/Truthy)

Reply:
right you are. sorted